pkg/controller/servicedns/controller.go

```diff
@@ -358,7 +354,7 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 	fedDNS.Status.Domain = domainObj.Domain
 
 	if !reflect.DeepEqual(cachedDNS.Status, fedDNS.Status) {
-		_, err = c.fedClient.MulticlusterdnsV1alpha1().ServiceDNSRecords(fedDNS.Namespace).UpdateStatus(fedDNS)
+		err = c.fedClient.Status().Update(context.TODO(), fedDNS)
```

Comment:
Why is `context.TODO` preferred over `context.Background`?

Reply:
while both are same, context.TODO reminds us that it needs a rework in the future to use the single context for the code path across the functions. it is not logical to use context.Background() everywhere. here is the comment on its usage:
```
// Background returns a non-nil, empty Context. It is never canceled, has no
// values, and has no deadline. It is typically used by the main function,
// initialization, and tests, and as the top-level Context for incoming
// requests.
```
```
// TODO returns a non-nil, empty Context. Code should use context.TODO when
// it's unclear which Context to use or it is not yet available (because the
// surrounding function has not yet been extended to accept a Context
// parameter). TODO is recognized by static analysis tools that determine
// whether Contexts are propagated correctly in a program.
```